kratos/geometries/line_2d_2.h

```diff
@@ -1086,28 +1086,18 @@ class Line2D2 : public Geometry<TPointType>
         const TPointType& r_first_point  = BaseType::GetPoint(0);
         const TPointType& r_second_point = BaseType::GetPoint(1);
 
-        // Project point
-        const double tolerance = 1e-14; // Tolerance
-
         const double length = Length();
 
-        const double length_1 = std::sqrt( std::pow(rPoint[0] - r_first_point[0], 2)
-                    + std::pow(rPoint[1] - r_first_point[1], 2));
+        // Project the point on the line in global space
+        const array_1d<double, 3> vector_from_first_point_to_input = rPoint - r_first_point;
+        const array_1d<double, 3> unity_line_direction             = (r_second_point - r_first_point) / length;
+        const auto projection_on_line                              = inner_prod(vector_from_first_point_to_input, unity_line_direction);
 
-        const double length_2 = std::sqrt( std::pow(rPoint[0] - r_second_point[0], 2)
-                    + std::pow(rPoint[1] - r_second_point[1], 2));
+        // Conversion to local space
+        constexpr double tolerance = 1e-14;
```

Comment:
i understand that this tolerance was hardcoded before, but it is definitely now too nice ( for example it may fail for very small geometries). 

Not telling that we should consider this as a blocker, but this is not a very robust solution.

Our point here is: why do we need the tolerance at all? anyhow you are assuming lenght to be different from zero, otherwise it would fail already in 1093

Reply:
Yes, I completely agree that we shouldn't have the tolerance here at all. I tried before to remove it, but the `IsInside` function seems to expect the results from `PointLocalCoordinates` to be as they are (as @loumalouomega mentioned), so tests started failing.

To keep the scope of this PR limited, I only fixed the projection itself, but kept the conversion of the projection to a $\xi$ value as is.

As proposed by @loumalouomega and @matekelemen, the IsInside function should be changed (i.e. the tolerance should be applied there instead of in `PointLocalCoordinates`), so I would propose investigate and change this in a separate small-scoped PR.